src/ucp/wireup/select.c

```diff
@@ -1850,6 +1871,34 @@ ucp_wireup_add_rma_bw_lanes(const ucp_wireup_select_params_t *select_params,
         bw_info.criteria.local_md_flags |= UCT_MD_FLAG_INVALIDATE_RMA;
     }
 
+    /* Find an AM lane that needs to be segregated from RMA accesses */
+    for (lane_desc_idx = 0; lane_desc_idx < select_ctx->num_lanes;
+         ++lane_desc_idx) {
+
+        if (!(select_ctx->lane_descs[lane_desc_idx].lane_types &
+              UCS_BIT(UCP_LANE_TYPE_AM))) {
+            continue;
+        }
+
+        rsc_index  = select_ctx->lane_descs[lane_desc_idx].rsc_index;
+        iface_attr = ucp_worker_iface_get_attr(ep->worker, rsc_index);
+
+        if (iface_attr->cap.flags & UCT_IFACE_FLAG_TX_RX_DEPENDENCY) {
+            am_lane = lane_desc_idx;
+
+            /* Map AM lane to specific path_index */
+            path_index = select_ctx->lane_descs[lane_desc_idx].path_index;
+            ucs_assertv((path_index == 0) ||
+                        (path_index == UCP_WIREUP_PATH_INDEX_UNDEFINED),
+                        "rsc_index=%u lane_desc_idx=%u path_index=%u",
+                        rsc_index, lane_desc_idx, path_index);
+
+            select_ctx->lane_descs[lane_desc_idx].path_index = 0;
+            bw_info.criteria.num_paths_override              = 1;
```

Comment:
IMO better add num_paths_override as a parameter to ucp_wireup_add_bw_lanes rather than add it to criteria struct, since it's relevant only when adding bw lanes

Reply:
fixed